RecoPixelVertexing/PixelTriplets/interface/CAHitQuadrupletGenerator.h

```diff
@@ -131,5 +131,10 @@ class CAHitQuadrupletGenerator {
     const float caThetaCut = 0.00125f;
     const float caPhiCut = 0.1f;
     const float caHardPtCut = 0.f;
+    std::vector<std::string> layerList;
+    edm::ParameterSetDescription BPix;
+    edm::ParameterSetDescription FPix;
```

Comment:
I think for each layer their psets have to be provided. For example, TTRHBuilder is used here https://github.com/cms-sw/cmssw/blob/master/RecoTracker/TkSeedingLayers/src/SeedingLayerSetsBuilder.cc#L101 and for proper TTRHBuilder acc. to FastSim, hitProducer will also change. Since these CA quadruplets & triplets iterations involve only BPix and FPix, that is why I have not included any other layer pset in the config files.

Reply:
I don't understand what a configuration has to do with these `BPix` and `FPix` members. As far as I can tell, these specific member variables are not used for anything. In general I'm not aware of any use case for `edm::ParameterSetDescription` members (`edm::ParameterSetDescription` is used only inside static `fillDescriptions()` method).

Reply:
I tried not giving the parameters BPix & FPix to the iteration configs but I got the error 
%MSG-e SeedingLayerSetsBuilder:   TrajectorySeedProducer:detachedQuadStepSeeds@ctor 17-Apr-2018 08:36:00 CEST  pre-events
configuration for layer: BPix1 not found, job will probably crash!
%MSG
----- Begin Fatal Exception 17-Apr-2018 08:36:00 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=TrajectorySeedProducer label='detachedQuadStepSeeds'
Exception Message:
MissingParameter: Parameter 'TTRHBuilder' not found.
----- End Fatal Exception -------------------------------------------------
coming from https://github.com/cms-sw/cmssw/blob/master/RecoTracker/TkSeedingLayers/src/SeedingLayerSetsBuilder.cc#L264
I think that's why these parameters are required. Although providing only TTRHBuilder in the pset for the layers is enough, hitProducer is not required for FastSim.

Reply:
I did not ask to remove any parameters from configuration, but these two member variables from `CAHitQuadrupletGenerator` class (as they are not used). You can keep the configuration as it is.

Reply:
Ah... sorry my bad! I misunderstood.. thanks.